### ISSUE TITLE
Dead kernel

### ISSUE THREAD

**Issue by carloslesmes** [JuliaLang/IJulia.jl#159]
After a few days I opened a note and found this message:

```
Dead kernel
The kernel has died, and the automatic restart has failed. It is possible the kernel cannot be restarted. If you are not able to restart the kernel, you will still be able to save the notebook, but running code will no longer work until the notebook is reopened.
```

```
Kernel Restarting
The kernel appears to have died. It will restart automatically.
```
Alternating...

I can't find a way to restart the kernel, please help.

**Comment by stevengj** [comment]
Seems no longer applicable to the latest version.

**Comment by luisjavierhernandez** [comment]
I have been working with my instance of JuliaBox with the kernel Julia 0.4.1 without problems, but
some days ago, a new Kernel Julia 0.4.2 has been installed. My problem is that in my instance of
JuliaBox this Kernel Julia 0.4.2 has died and it is not possible for me to restart this kernel. Therefore I have returned to Kernel Julia 0.3 , but this a problem because I have to modify the code of many functions.

Any idea to solve my problem?

Thanks 

Luis Javier Hernandez

**Comment by tanmaykm** [comment]
@luisjavierhernandez this is most likely because of some conflicting locally installed/precompiled packages. Clearing them from ~/.julia should resolve the issue, e.g.:
````mv ~/.julia ~/oldpkgs````

There's a button on the JuliaBox settings tab to do it as well.

**Comment by luisjavierhernandez** [comment]
Thanks Tanmay,

Your instruction is correct.
When I remove the packages the Kernel of Julia 0.4.2  works 
properly. 

However, my program uses PyPlot and this package 
needs PyCall. The problem appears again when I try 
to install the package PyCall. 

If I work with the Kernel Julia 0.3 
I do not have any problems with the installation of PyPlot and PyCall.

Is it possible to install PyCall without  having
conflicts with the Kernel of Julia 0.4.2?

With my best wishes

**Comment by stevengj** [comment]
I think this is related to stevengj/PyPlot.jl#185, which in turn seems like some underlying problem with Julia's caching mechanism on JuliaBox, nothing specific to IJulia or PyPlot per se.

**Comment by stevengj** [comment]
In particular, I see a clearer error message in the JuliaBox console:
```
juser@juliabox:~$ /opt/julia-0.4.2/bin/julia                                                                                                                       
INFO: Recompiling stale cache file /opt/julia_packages/.julia/lib/v0.4/Compat.ji for module Compat.                                                                
               _
   _       _ _(_)_     |  A fresh approach to technical computing
  (_)     | (_) (_)    |  Documentation: http://docs.julialang.org
   _ _   _| |_  __ _   |  Type "?help" for help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 0.4.2 (2015-12-06 21:47 UTC)
 _/ |\__'_|_|_|\__'_|  |  Official http://julialang.org/ release
|__/                   |  x86_64-unknown-linux-gnu
                                                                                                                                                                   
julia> using IJulia                                                                                                                                                
INFO: Precompiling module IJulia...                                                                                                                                
INFO: Recompiling stale cache file /opt/julia_packages/.julia/lib/v0.4/Compat.ji for module Compat.                                                                
ERROR: LoadError: unlink: read-only file system (EROFS)                                                                                                            
 in unlink at fs.jl:102                                                                                                                                            
 in rm at file.jl:59                                                                                                                                               
 in create_expr_cache at loading.jl:330                                                                                                                            
 in recompile_stale at loading.jl:461                                                                                                                              
 in _require_from_serialized at loading.jl:83                                                                                                                      
 in _require_from_serialized at ./loading.jl:109                                                                                                                   
 in require at ./loading.jl:219                                                                                                                                    
 in include at ./boot.jl:261                                                                                                                                       
 in include_from_node1 at ./loading.jl:304                                                                                                                         
 [inlined code] from none:2                                                                                                                                        
 in anonymous at no file:0                                                                                                                                         
 in process_options at ./client.jl:257                                                                                                                             
 in _start at ./client.jl:378                                                                                                                                      
while loading /opt/julia_packages/.julia/v0.4/IJulia/src/IJulia.jl, in expression starting on line 4                                                               
ERROR: Failed to precompile IJulia to /home/juser/.julia/lib/v0.4/IJulia.ji                                                                                        
 in error at ./error.jl:21                                                                                                                                         
 in compilecache at loading.jl:384                                                                                                                                 
 in require at ./loading.jl:250                                                                                                                                    
```

**Comment by stevengj** [comment]
See the abovementioned Julia issue.

**Comment by stevengj** [comment]
In general, continuing to comment on this issue is not fruitful.   Not all causes of "dead kernel" messages are created equal — each issue should be tied to a specific problem in IJulia (or, in this case, in Julia and JuliaBox).

**Comment by samirOs** [comment]
I had a similar issue on centos, however, the issue was coming from a bad packaging of IJulia, as i didn't have cmake so i saw the error during the Pkg.add("IJulia"). 
the solution was to make sure to yum install cmake 
then 
do Pkg.build("IJulia")
this will build IJulia, make sure there are no errors.
Once this is ok, try launching your notebook and check

**Comment by saurabh2804** [comment]
I had the same issue on Manjaro linux recently. I was using the version available in the Manajaro repository (1.01). I simply uninstalled it and downloaded Julia from the Julia website, extracted it and added it to my path. It is working without any problem. Installed IJulia and using it with Jupyter notebook.
